src/ngraph/runtime/reference/concat.hpp

```diff
@@ -32,12 +32,15 @@ namespace ngraph
                         T* out,
                         const std::vector<Shape>& in_shapes,
                         const Shape& out_shape,
-                        size_t concatenation_axis)
+                        int64_t concatenation_axis)
             {
                 // We will copy the inputs to the output one at a time. As we go, we will move out
                 // along the concatenation axis, starting at 0.
-                size_t concatenation_pos = 0;
 
+                concatenation_axis = concatenation_axis < 0
+                                         ? concatenation_axis + int64_t(in_shapes.size())
```

Comment:
I mad a mistake in my last comment this is the correct way.
change in_shapes.size() to out_shape.size()

Reply:
I will do fallback to the previous version. Thanks.